src/main/kotlin/open/tresorier/services/OperationService.kt

```diff
@@ -123,7 +134,56 @@ class OperationService(private val operationDao: IOperationDao, private val auth
         if (type == "DEBIT") {
             amount *= -1
         }
-        val operationCreated = Operation(account.id, day, null, amount,Time.now(), memo, false, false)    // créer une opération sans la mettre dans la base de donnée
+        val operationCreated = Operation(account.id, day, null, amount,Time.now(), memo, false, false, null)
         return operationCreated
     }
+    fun findDaughterOperations(person: Person, motherOperation: Operation): List<Operation> {
+        val motherOperationId = motherOperation.id
+        val account = accountDao.getById(motherOperation.accountId)
+        authorizationService.cancelIfUserIsUnauthorized(person, account)
+        val listAllOperations: List<Operation> = operationDao.findByAccount(account, null)
+        var listDaughterOperation = mutableListOf<Operation>()
+        listAllOperations.forEach {
+            if (it.motherOperationId == motherOperationId) {
+                listDaughterOperation.add(it)
+            }
+        }
+        return listDaughterOperation
+    }
+    fun findMotherOperationsByAccount(person: Person, account: Account, category: Category?) : List<Operation> {
+        val listAllOperations = this.findByAccount(person, account, category)
+        var listMotherOperations = mutableListOf<Operation>()
+        listAllOperations.forEach {
+            if (it.motherOperationId == null) {
+                listMotherOperations.add(it)
+            }
+        }
+        return listMotherOperations
+    }
+    fun findMotherOperationByDaugtherOperation(person: Person, daughterOperation: Operation) : Operation? {
+        if (daughterOperation.motherOperationId != null) {
+            val account = accountDao.getById(daughterOperation.accountId)
+            val motherOperationID = daughterOperation.motherOperationId
+            authorizationService.cancelIfUserIsUnauthorized(person, account)
+            val listAllOperations: List<Operation> = operationDao.findByAccount(account, null)
+            var motherOperation: Operation? = null 
+            listAllOperations.forEach {
+                if (motherOperationID == it.id) {
+                    motherOperation = it
+                }
+            }
+            return motherOperation
+        }
+        return null
+    }
+    fun findAllDaughterOperations(person: Person, account: Account, category: Category?) : List<Operation> {
+        val listAllOperations = this.findByAccount(person, account, category)
```

Comment:
the search should be conducted by the dao instead of the service

Reply:
removed because is a dead code